src/core/topology.cpp

```diff
@@ -26,99 +26,62 @@
  *  For more information see topology.hpp
  *   */
 
-#include "utils.hpp"
 #include "topology.hpp"
-#include "statistics_chain.hpp"
-#include "particle_data.hpp"
-#include "cells.hpp"
-#include "communication.hpp"
 #include "molforces.hpp"
+#include "particle_data.hpp"
 
-int     n_molecules = -1;
-Molecule *topology = nullptr;
+std::vector<Molecule> topology;
 int topo_part_info_synced = 0;
 
-void realloc_topology(int size)
-{
-  int m;
-
-  for(m = size ; m < n_molecules; m++) {
-    realloc_intlist(&topology[m].part, 0);
-  }
-  
-  topology = Utils::realloc(topology, size*sizeof(Molecule));
-
-  if (n_molecules < 0)
-    n_molecules = 0;
-  for(m = n_molecules; m < size; m++) {
-    init_intlist(&topology[m].part);
-#ifdef MOLFORCES
-    topology[m].trap_flag = 32; 
-    topology[m].noforce_flag = 32;
-    topology[m].favcounter = -1;
-    topology[m].fav[0] = 0;
-    topology[m].fav[1] = 0;
-    topology[m].fav[2] = 0;
-    topology[m].trap_force[0] = 0;
-    topology[m].trap_force[1] = 0;
-    topology[m].trap_force[2] = 0;
-#endif /*MOLFORCES*/
-  }
-  n_molecules = size;
-  
+void realloc_topology(int size) {
+  topology.resize(size);
   topo_part_info_synced = 0;
-
 }
 
 // Parallel function for synchronising topology and particle data
 void sync_topo_part_info() {
-  int i,j;
-  Particle* p;
-  for ( i = 0 ; i < n_molecules ; i ++ ) {
-    for ( j = 0 ; j < topology[i].part.n ; j++ ) {
-      p = local_particles[topology[i].part.e[j]];
-      if(!p) { 
-	/* Do nothing */ 
-      } 
-      else {
-	p->p.mol_id = i;
+  int i, j;
+  Particle *p;
+
+  for (unsigned molid = 0; molid < topology.size(); molid++) {
+    auto const &mol = topology.at(molid);
+    for (auto const &pid : mol.part) {
+      auto p = local_particles[pid];
+
+      if (p) {
+        p->p.mol_id = molid;
       }
     }
   }
 
   topo_part_info_synced = 1;
-
 }
 
-
-int set_molecule_trap(int mol_num, int trap_flag,DoubleList *trap_center,double spring_constant, double drag_constant, int noforce_flag, int isrelative) {
+int set_molecule_trap(int mol_num, int trap_flag, DoubleList *trap_center,
```

Comment:
Could not find any places where this method is actually used..
No interface yet?

Reply:
That's not interfaced but we decided to keep it for now.